src/wallet/wallet.cpp

```diff
@@ -1440,15 +1440,15 @@ void CWallet::transactionAddedToMempool(const CTransactionRef& tx) {
     }
 }
 
-void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRemovalReason reason) {
+void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, const MemPoolRemovalReason& reason) {
     LOCK(cs_wallet);
     auto it = mapWallet.find(tx->GetHash());
     if (it != mapWallet.end()) {
         RefreshMempoolStatus(it->second, chain());
     }
     // Handle transactions that were removed from the mempool because they
     // conflict with transactions in a newly connected block.
-    if (reason == MemPoolRemovalReason::CONFLICT) {
+    if (std::get_if<ConflictReason>(&reason)) {
```

Comment:
Could replace all the `std::get_if` for a more readable `IsReason<Conflict>(&reason)` if you include this function:
```c++
template <typename T>
bool IsReason(const MemPoolRemovalReason& reason)
{
    return std::get_if<T>(reason) != nullptr;
}
```

Reply:
Thanks @furszy